homeassistant/components/lutron/fan.py

```diff
@@ -0,0 +1,99 @@
+"""Lutron fan platform."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pylutron import Output
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN, LutronData
+from .entity import LutronDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Lutron fan platform.
+
+    Adds fan controls from the Main Repeater associated with the config_entry as
+    fan entities.
+    """
+    entry_data: LutronData = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            LutronFan(area_name, device, entry_data.client)
+            for area_name, device in entry_data.fans
+        ],
+        True,
+    )
+
+
+class LutronFan(LutronDevice, FanEntity):
+    """Representation of a Lutron fan."""
+
+    _attr_should_poll = False
+    _attr_speed_count = 3
+    _attr_supported_features = FanEntityFeature.SET_SPEED
+    _lutron_device: Output
+    _prev_percentage: int | None = None
+
+    @property
+    def percentage(self) -> int | None:
+        """Return the current speed."""
+        new_percentage = self._lutron_device.last_level()
```

Comment:
Move this to `update` instead and set `self._attr_percentage`.

Reply:
I see this isn't a polling entity. Then I think we should have a new callback method on the entity to update state and write it to the state machine.

Reply:
We can implement `_update_callback` and then call `super()._update_callback` inside after updating state attributes.

Reply:
when would this fire exactly? the integration can lose state quite easily which is why it was implemented this way from day one.  this is a major change to the functionality and could have significant impact to systems.  this is the way all of the entities function.  I'm not super inclined to change that for this platform.